test/common/buffer/buffer_test.cc

```diff
@@ -1075,6 +1076,33 @@ TEST(BufferHelperTest, WriteBEI64) {
     EXPECT_EQ("\x7F\xFF\xFF\xFF\xFF\xFF\xFF\xFF", buffer.toString());
   }
 }
+TEST(BufferHelperTest, AddFragments) {
+  {
+    // Add some string fragments.
+    Buffer::OwnedImpl buffer;
+    buffer.addFragments({"aaaaa", "bbbbb"});
+    EXPECT_EQ("aaaaabbbbb", buffer.toString());
+  }
+
+  {
+    // Add string fragments cyclically.
+    Buffer::OwnedImpl buffer;
+    std::string str;
+    for (size_t i = 0; i < 1024; i++) {
+      buffer.addFragments({"aaaaa", "bbbbb", "ccccc", "ddddd"});
+      str += "aaaaabbbbbcccccddddd";
+    }
+    EXPECT_EQ(20 * 1024, buffer.length());
+    EXPECT_EQ(str, buffer.toString());
+
+    auto slice_vec = buffer.getRawSlices();
+
+    EXPECT_EQ(slice_vec.size(), 5);
```

Comment:
likewise here as in https://github.com/envoyproxy/envoy/pull/19115/files#r780540013

Reply:
get it.